#161342841 Create route to post articles

victor-nweze-deel opened the pull request:
#### What does this PR do?
Create a route to  post articles

#### Description of Task to be completed?
- Create an article model
- Refactor migrations to include associations
- Create article repository
- Create article route
- Write unit and integration tests

#### How should this be manually tested?
- Git clone the repository
- Cd into the the repo folder
- Run `git checkout ft-create-articles-161342841`

- Open up postman and hit the following `localhost:5000` route with the json input
```
{
   title: "my first title",
   body: "this is the body of the article",
   description: "this is the articles description",
   images: "https://imgur.com/121212kjhkjjhbj.jpg"
}
```
- Run `npm test`

#### What are the relevant pivotal tracker stories?
[#161342841](https://www.pivotaltracker.com/story/show/161342841)

#### Screenshots (if appropriate)
<img width="1139" alt="screen shot 2018-10-29 at 7 09 15 am" src="https://user-images.githubusercontent.com/41582565/47635264-ac583d80-db54-11e8-8604-3a397eb50777.png">
<img width="906" alt="screen shot 2018-10-27 at 10 00 35 am" src="https://user-images.githubusercontent.com/41582565/47635275-b2e6b500-db54-11e8-8a40-8ccf8d5dce75.png">

wiztemple commented:
The readTime is 57? 57s or 57 minutes?

wiztemple commented:
the number of users up to 222? maybe a test database will be cool for it.

victor-nweze-deel commented:
> The readTime is 57? 57s or 57 minutes?

The read-time is in seconds. Implementing the logic to convert the integer to hours, minutes or seconds is unnecessary at the back end.

victor-nweze-deel commented:
> the number of users up to 222? maybe a test database will be cool for it.

Noted! Nice catch

ucheg6 commented:
Good job Victor...This looks great but we can import all the routers in the index file and export it again so we can use it in app.js  instead of having multiple routes in app.js file.

TheoOkafor commented:
> Good job Victor...This looks great but we can import the all routers in the index file and export it again so we can use it in app.js instead of having multiple routes in app.js file.

Great suggestion Uche

TheoOkafor commented:
It looks good. Good job Victor

seyi-adeleke commented on app.js:
```diff
@@ -20,7 +26,9 @@ app.use(logger('dev'));
 app.use(cleanStrings);
-app.use('/api/v1', router);
+
+app.use('/api/articles', articleRouter);
```
This isn't very clean to me.  We should only be exporting one instance of the router into this file. We should have an index router file that encapsulates all the endpoints for our entities. Something like this. 
```
export default (router) => {
    article(router);
    user(router)
};
```
Where article and user are instances of the article and user routes. Let me know if you have questions

victor-nweze-deel replied:
Noted

seyi-adeleke commented on server/controllers/article.js:
```diff
@@ -0,0 +1,39 @@
+      description,
+      images,
+    } = request.body;
+    const readtime = Math.floor(body.split(' ').length / 3);
```
why 3?

victor-nweze-deel replied:
From my research, the average person reads 180 words per minute on a monitor https://en.wikipedia.org/wiki/Words_per_minute. So that's 3 words per second on the average.

seyi-adeleke commented on server/models/articles.js:
```diff
@@ -0,0 +1,50 @@
+
+module.exports = (sequelize, DataTypes) => {
+  const Articles = sequelize.define('Articles', {
+    id: {
```
Thought we agreed to not use UUIDs? It should be consistent with what we have in the user model

ProfJigsaw replied:
Actually, we agreed to use UUIDs but there was an oversight during the creation of User model.

victor-nweze-deel replied:
Feedback received

seyi-adeleke commented on server/repository/articleRepository.js:
```diff
@@ -0,0 +1,40 @@
+        slug: article.slug,
+      }
+    });
+    if (!articleRecord) return null;
```
No one line if statements please

victor-nweze-deel replied:
Noted

seyi-adeleke commented on server/routes/article.js:
```diff
@@ -0,0 +1,9 @@
+
+const router = new Router();
+
+router.post('/', jwtAuth, Article.createArticle);
```
jwtAuth?

seyi-adeleke replied:
how about isAuthenticated?

victor-nweze-deel replied:
Noted

seyi-adeleke commented on server/routes/index.js:
```diff
@@ -0,0 +1,7 @@
+
+export default (app) => {
+  app.use('/api/v1', userRouter);
+  app.use('/api/articles', articleRouter);
```
Shouldn't this be `/api/v1/` as well?

wiztemple committed:
ft(articles): Create an article

- Create an article model
- Refactor migrations to include associations
- Create article repository
- Create article route
- Write unit and integration tests

victor-nweze-deel committed:
ft(articles): Create a new article
